addons/account/tests/test_ir_actions_report.py

```diff
@@ -96,3 +97,141 @@ def test_download_one_encrypted_pdf(self):
         # trying to merge with a corrupted attachment should not work
         with self.assertRaises(UserError):
             self.env['ir.actions.report'].with_context(force_report_rendering=True)._render_qweb_pdf('account.action_account_original_vendor_bill', res_ids=[in_invoice_1.id, in_invoice_2.id])
+
+    def test_report_with_some_resources_reloaded_from_attachment(self):
+        """
+        Test for opw-3827700, which caused reports generated for multiple invoices to fail if there was an invoice in
+        the middle that had an attachment, and 'Reload from attachment' was enabled for the report. The misbehavior was
+        caused by an indexing issue.
+        """
+        first_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Something',
+                'quantity': 1,
+                'price_unit': 123,
+                'product_id': self.product_a.id
+            })]
+        })
+
+        second_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Something',
+                'quantity': 1,
+                'price_unit': 123,
+                'product_id': self.product_a.id
+            })]
+        })
+
+        third_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': self.partner_a.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Something',
+                'quantity': 1,
+                'price_unit': 123,
+                'product_id': self.product_a.id
+            })]
+        })
+
+        # Post invoices to be able to associate attachments.
+        for invoice in [first_invoice, second_invoice, third_invoice]:
+            invoice.action_post()
+
+        invoices_report_ref = 'account.report_invoice_with_payments'
+
```

Comment:
`variable_name = self.env['ir.actions.report'].with_context(force_report_rendering=True)`

This is part of a **suggestion** to make it nicer to read.